microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/commons/JavaCursorContext.java

```diff
@@ -0,0 +1,31 @@
+/*******************************************************************************
+* Copyright (c) 2023 Red Hat Inc. and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+* which is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lsp4mp.commons;
+
+/**
+ * Represents the context where the cursor is in a Java file.
+ */
+public enum JavaCursorContext {
+
+	EMPTY_FILE(1),
+	CLASS(2),
```

Comment:
I will remove this enum and add:

 * `ON_BODY_CLASS` : triggered on a body class, but not on a body method
 * `ON_ANNOTATION_CLASS` : triggered before on a class where annotation can be inserted.
 *  `ON_ANNOTATION_METHOD` : triggered before on a method where annotation can be inserted.
 *  `ON_ANNOTATION_FIELD`:  triggered before on a field where annotation can be inserted.

Reply:
Sorry, I don't understand the `ON_ANNOTATION_CLASS`, `ON_ANNOTATION_METHOD`, `ON_ANNOTATION_FIELD`. If you are in a body class, then aren't you always before a field, a method, or a class? I think that all the annotations that we have snippets for are for methods.

Reply:
...unless you are in a class with nothing in the body :facepalm:. Okay makes sense.

Reply:
Let me know what you think of the new naming scheme

Reply:
You can also annotate parameters of a method declaration. Is it worth adding that ? I think only Qute has an example of that.

Reply:
Unless we are planning on adding a snippet that supports a specific annotation in the parameters, I don't think it's worthwhile. What is the example in Qute?

Reply:
https://github.com/redhat-developer/quarkus-ls/issues/637 . Probably not something we'd be looking into for the time being.

Reply:
Yeah, since it's just a single member annotation, I don't think it's worth making a snippet for